.github/workflows/performance.yml

```diff
@@ -162,8 +199,32 @@ jobs:
       - name: Run baseline performance tests
         run: npm run test:performance -- --prefix=base
 
-      - name: Print base line performance tests results
-        run: "node ./tests/performance/results.js --prefix=base"
+      - name: Print baseline performance tests results
+        run: node ./tests/performance/results.js --prefix=base
+
+      - name: Compare results with base
+        run: node ./tests/performance/compare-results.js ${{ runner.temp }}/summary.md
+        env:
+          TARGET_SHA: ${{ github.event_name == 'pull_request' && github.event.pull_request.base.sha || github.event.before }}
+
+      - name: Add workflow summary
+        run: cat ${{ runner.temp }}/summary.md >> $GITHUB_STEP_SUMMARY
+
+      - name: Check if a comment was already made
+        id: find-comment
+        uses: peter-evans/find-comment@a54c31d7fa095754bfef525c0c8e5e5674c4b4b1
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body-includes: Performance test results for
+
+      - name: Comment on PR with test results
+        uses: peter-evans/create-or-update-comment@c6c9a1a66007646a28c153e2a8580a5bad27bcfa
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-id: ${{ steps.find-comment.outputs.comment-id }}
+          edit-mode: replace
+          body-path: ${{ runner.temp }}/summary.md
+          token: ${{ secrets.PR_COMMENT_TOKEN }}
```

Comment:
Per my feedback on Slack in the performance chat, can we pull that out into a separate ticket and PR? While it's technically certainly good, the metrics are too unstable at this point to post them _that obviously_ on every PR. People that aren't familiar with the metrics work we've been doing (which certainly is the majority of contributors) may over-interpret a slightly positive or slightly negative result, not knowing that it just comes from variance.

In the interest of time, I think we should merge the rest and reconsider this part for a Future Release.
```suggestion
```